### PR TITLE
Refactor auxiliary functions inside of watson.py to watson_util.py to allow for future expandability

### PR DESCRIPTION
**Pull request by ZachEichen**
Given some future plans that involve outputs from Watson cloud api's I determined that there are a number of common tasks required between many of them. A lot of these tasks have already been addressed in the watson.py sub-module of the io module. 

Therefore I refactored those tasks into a separate sub-module to allow for future expandability, which would allow future projects to more easily interface with Watson cloud api outputs.

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -49,7 +49,8 @@
 # This pattern also affects html_static_path and html_extra_path.
 
 
-exclude_patterns = ['_build', 'Thumbs.db', '.DS_Store','text_extensions_for_pandas.util.rst']
+exclude_patterns = ['_build', 'Thumbs.db', '.DS_Store','text_extensions_for_pandas.util.rst',
+                    'text_extensions_for_pandas.io.watson_util.rst']
```
[the remark this answers is not in the document]
Comment (ZachEichen) [reply docs/conf.py:53]: yes this is on purpose.

File: text_extensions_for_pandas/io/watson_util.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
+#
```
[the remark this answers is not in the document]
Comment (ZachEichen) [reply text_extensions_for_pandas/io/watson_util.py:1]: I just checked. there doesn't seem to be any such tests.

File: text_extensions_for_pandas/io/watson.py [hunk 3]
```diff
@@ -32,6 +32,7 @@
 import pyarrow as pa
 
 from text_extensions_for_pandas.array import CharSpanArray, TokenSpanArray
+import text_extensions_for_pandas.io.watson_util as util
```
[the remark this answers is not in the document]
Comment (ZachEichen) [reply text_extensions_for_pandas/io/watson.py]: should be fixed in the latest commit.

**Comment by frreiss** [comment]
Changes look good, but there are some conflicts with the master branch. @ZachEichen can you fix those merge conflicts so I can merge this PR, please?

**Pull request merged**